Sources/DeepLearning/Loss.swift

```diff
@@ -16,14 +16,38 @@
 import TensorFlow
 #endif
 
+/// Computes the mean squared error between logits and labels.
+///
+/// - Parameters
+///   - logits: one-hot encoded outputs from a neural network.
+///   - labels: one-hot encoded values that correspond to the correct output.
 @differentiable
 public func meanSquaredError<Scalar: TensorFlowFloatingPoint>(
     predicted: Tensor<Scalar>, expected: Tensor<Scalar>) -> Tensor<Scalar> {
     return (expected - predicted).squared().mean()
 }
 
+/// Computes the softmax cross entropy (categorical cross entropy) between logits and labels.
+///
+/// - Parameters
+///   - logits: one-hot encoded outputs from a neural network.
+///   - labels: one-hot encoded values that correspond to the correct output.
 @differentiable
 public func softmaxCrossEntropy<Scalar: TensorFlowFloatingPoint>(
     logits: Tensor<Scalar>, labels: Tensor<Scalar>) -> Tensor<Scalar> {
     return -(labels * logSoftmax(logits)).mean(alongAxes: 0).sum()
 }
+
+/// Computes the sigmoid cross entropy (binary cross entropy) between logits and labels.
+///
+/// - Parameters
+///   - logits: single continuous values from 0 to 1.
+///   - labels: integer values that correspond to the correct output.
+@differentiable
+public func sigmoidCrossEntropy<Scalar: TensorFlowFloatingPoint>(
+    logits: Tensor<Scalar>, labels: Tensor<Scalar>) -> Tensor<Scalar> {
+    let loss = labels * log(logits) +
+               (Tensor<Scalar>(ones: labels.shape) - labels) *
```

Comment:
The call to `init(ones:)` is unnecessary and inefficient. Since `-` is broadcasting, yon can just use `1 -`. The whole statement can be simplified to:
```swift
let loss = labels * log(logits) + (1 - labels) * log(1 - logits)
```

Reply:
I did try that at first, but got this error:

```
error: repl.swift:23:2: error: function is not differentiable
@differentiable
~^~~~~~~~~~~~~~

repl.swift:24:13: note: when differentiating this function definition
public func sigmoidCrossEntropy<Scalar: TensorFlowFloatingPoint>(
            ^

repl.swift:26:42: note: function call is not differentiable because generic requirements are not met
    let loss = labels * log(logits) + (1 - labels) * log(1 - logits)
                                         ^


```

Reply:
Oh, in that case it's definitely a high-priority bug.  Could you file a bug on bugs.swift.org for this as well?

Reply:
Could you also try:
```swift
let loss = labels * log(logits) + (Scalar(1) - labels) * log(Scalar(1) - logits)
```
?

Reply:
Same error (had to split it up into chunks because `the compiler is unable to type-check this expression in reasonable time`):

```
error: repl.swift:23:2: error: function is not differentiable
@differentiable
~^~~~~~~~~~~~~~

repl.swift:24:13: note: when differentiating this function definition
public func sigmoidCrossEntropy<Scalar: TensorFlowFloatingPoint>(
            ^

repl.swift:28:29: note: function call is not differentiable because generic requirements are not met
        let p2 = (Scalar(1) - labels)
                            ^


```

Reply:
Filed the bug report:

Reply:
This does work:

```
@differentiable
public func sigmoidCrossEntropy<Scalar: TensorFlowFloatingPoint>(
    logits: Tensor<Scalar>, labels: Tensor<Scalar>
) -> Tensor<Scalar> {
    let loss = labels * log(logits) +
        (Tensor<Scalar>(1) - labels) *
        log(Tensor<Scalar>(1) - logits)
    return -loss.mean(alongAxes: 0).sum()
}
```

Reply:
This looks good, please use that for now. You can move `log(Tensor<Scalar>(1) - logits)` to the end of the second line of the statement. We are enforcing a horizontal limit of 100 columns.

Reply:
Done. Thanks!